cmds/cpud/serve.go

```diff
@@ -91,14 +95,20 @@ func listen(network, port string) (net.Listener, error) {
 			return nil, err
 		}
 		ln, err = vsock.ListenContextID(any, uint32(p), nil)
-
+		if err != nil {
```

Comment:
This return is not needed. 

There seem to be several different things going on in this PR, can you split out into multiple PRs. It makes bisect way easier.